samtranslator/model/__init__.py

```diff
@@ -489,7 +489,11 @@ def __init__(self, *modules):
                 self.resource_types[resource_class.resource_type] = resource_class
 
     def can_resolve(self, resource_dict):
-        if not isinstance(resource_dict, dict) or "Type" not in resource_dict:
+        if (
+            not isinstance(resource_dict, dict)
+            or "Type" not in resource_dict
+            or not isinstance(resource_dict["Type"], string_types)
```

Comment:
nit, these two lines can be combined as `not isinstance(resource_dict.get('Type'), string_types)`